### PR TITLE
Merchant item index#17

### LINKED ISSUE #17: User Story 50, Merchant Items Index Page
```
As a merchant
When I visit my items page "/dashboard/items"
I see a link to add a new item to the system
I see each item I have already added to the system, including:
- the ID of the item
- the name of the item
- a thumbnail image for that item
- the price of that item
- my current inventory count for that item
- a link or button to edit the item

If no user has ever ordered this item, I see a link to delete the item
If the item is enabled, I see a button or link to disable the item
If the item is disabled, I see a button or link to enable the item
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #18

depends on #31

### PR DESCRIPTION
**Pull request by Bradniedt**
-Adds test for merchant index page. Merchant should see select item info, as well as additional links to enable/disable/edit. 
-Implements functionality to complete card 17
-All tests passing
-99.95% coverage

**NOTE - I did not test or build out the merchant item enable/disable/delete buttons because those are other cards that have to come after this card.** 

closes #17

### CODE REVIEW

File: app/views/items/index.html.erb [hunk 1]
```diff
@@ -21,6 +24,19 @@
...
+    <% if current_default? || !(current_user) %>
+      <%= button_to 'Add item', carts_path(item_id: item.id) %>
+    <% end %>
+    <% if current_merchant? %>
+      <%= link_to 'Edit this item', dashboard_item_edit_path(item) %>
+      <% unless !(item.order_items) %>
+        <%= link_to 'Delete this item', dashboard_item_path(:id => item.id), method: :delete %>
+      <% end %>
+      <% if item.enabled == true %>
+        <%= link_to 'Disable this item', dashboard_item_edit_path(item, :enable => true) %>
```
Comment (danbriechle) [inline app/views/items/index.html.erb]: You don't need to pass this enabled param and you don't want to.

File: app/views/items/index.html.erb [hunk 2]
```diff
@@ -21,6 +24,19 @@
     <p><%= item.description %></p>
     <p><%= item.inventory %></p>
     <p><%= item.current_price %></p>
-    <%= button_to 'Add item', carts_path(item_id: item.id) %>
+    <% if current_default? || !(current_user) %>
+      <%= button_to 'Add item', carts_path(item_id: item.id) %>
+    <% end %>
+    <% if current_merchant? %>
+      <%= link_to 'Edit this item', dashboard_item_edit_path(item) %>
+      <% unless !(item.order_items) %>
```
[the remark this answers is not in the document]
Comment (asmolentzov) [reply app/views/items/index.html.erb]: Ah! That makes sense why just a general `if` or `unless item.order_items` didn't work then - the 'empty' value is an empty array, which is still truthy! Looks good!

File: app/views/items/index.html.erb [hunk 3]
```diff
@@ -21,6 +24,19 @@
...
+        <%= link_to 'Delete this item', dashboard_item_path(:id => item.id), method: :delete %>
```
[the remark this answers is not in the document]
Comment (asmolentzov) [reply app/views/items/index.html.erb]: I just meant passing it in to the path: `dashboard_item_path(item)` like so. I agree that we still need the logic checking to make sure there are not order_items that include it!

**Comment by danbriechle** [comment]
Looks good to me! Approved
